docs/source/quickstart.rst

```diff
@@ -141,7 +141,46 @@ To start a dask cluster on YARN
                                        channels=['conda-forge'])
 
    # each worker gets 4GB and two cores
-   cluster.start(nworkers=10, memory=4096, cpus=2)
+   cluster.start(n_workers=10, memory=4096, cpus=2)
 
    from dask.distributed import Client
    client = Client(cluster)
+
+This starts a YARN cluster and creates the Client in the same process. Normally, one would use a
+separate script that would create the cluster with scheduler and workers, and sit around waiting
+for tasks. This script would run in its own console session. Other jobs could then use the
+started scheduler address to run tasks. Example:
```

Comment:
Interesting.  I have to admit that I rarely (actually never) use YARN, but this workflow seems odd to me.  What is the advantage of separating the process that manages the scheduler and workers from the process that the user interacts with?  I can certainly imagine cases where this would be advantageous (client is on a different network, clients come and go more frequently than scheduler/workers) but I'm curious why this type of workflow is more common on YARN systems than otherwise?

Reply:
This workflow seems odd to me also! This is how I was advised to do things when I started asking questions in https://github.com/dask/knit/issues/124. I started with Client and Scheduler in the same process. When things terminate, the Client spits out a ton of errors, which are scary to see as part of normal application flow. I don't want those things in my logs. So I was advised to use the long-running Scheduler.

Reply:
In case of Spark Yarn, executor containers are started on submitting the application, and they terminate right after.

Reply:
I'm guessing the right answer there might have been suppressing or redirecting those logs, as opposed to separating the scheduler? :)

Reply:
Heh, noted.  Honestly I think that the real answer to your questions might be the following:

Dask doesn't support Yarn well today.  Users will probably be *much* happier on just about *any other* resource manager.  This is due to two reasons:

1.  Yarn lacks friendly non-JVM APIs, and so is a pain to work with from Python
2.  Most Dask users have either been on newer systems (like Kubernetes) or non-JVM-centric systems like PBS, SLURM, SGE, LSF, or others.

However, there are active efforts to improve YARN support over at [Skein](https://github.com/jcrist/skein)

Reply:
> I'm guessing the right answer there might have been suppressing or redirecting those logs, as opposed to separating the scheduler? :)

Suppressing would work, sure.  It would also be nice to investigate why communications are being severed.  Given the ongoing rewrite though my inclination would be to just wait to see if all of these problems go away.

Reply:
If your objective is to get Dask running on cloud infrastructure I strongly recommend using [Kubernetes and Helm](http://dask.pydata.org/en/latest/setup/kubernetes-helm.html).  This is far more robust and far simpler to get going.

Reply:
Thank you! I'm going to throw  away most of the things in this PR, not to lead other people astray, and just leave the `nworkers` fix.